etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/messages/SendMessageHelperTest.groovy

```diff
@@ -0,0 +1,104 @@
+package gov.hhs.cdc.trustedintermediary.etor.messages
+
+import gov.hhs.cdc.trustedintermediary.context.TestApplicationContext
+import gov.hhs.cdc.trustedintermediary.etor.metadata.partner.PartnerMetadataException
+import gov.hhs.cdc.trustedintermediary.etor.metadata.partner.PartnerMetadataMessageType
+import gov.hhs.cdc.trustedintermediary.etor.metadata.partner.PartnerMetadataOrchestrator
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger
+import spock.lang.Specification
+
+class SendMessageHelperTest extends Specification {
+
+    def mockOrchestrator = Mock(PartnerMetadataOrchestrator)
+    def mockLogger = Mock(Logger)
+
+    def setup() {
+        TestApplicationContext.reset()
+        TestApplicationContext.init()
+        TestApplicationContext.register(SendMessageHelper, SendMessageHelper.getInstance())
+        TestApplicationContext.register(PartnerMetadataOrchestrator, mockOrchestrator)
+        TestApplicationContext.register(Logger, mockLogger)
+        TestApplicationContext.injectRegisteredImplementations()
+    }
+    def "savePartnerMetadataForReceivedMessage works"() {
+        given:
+        def hashCode = new Random().nextInt()
+        def messageType = PartnerMetadataMessageType.RESULT
+        def receivedSubmissionId = "receivedId"
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        SendMessageHelper.getInstance().savePartnerMetadataForReceivedMessage(receivedSubmissionId, hashCode, messageType)
+
+        then:
+        1 * mockOrchestrator.updateMetadataForReceivedMessage(_, _, _)
+    }
+
+    def "savePartnerMetadataForReceivedMessage should log warnings for null receivedSubmissionId"() {
+        given:
+        def hashCode = new Random().nextInt()
+        def messageType = PartnerMetadataMessageType.RESULT
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        SendMessageHelper.getInstance().savePartnerMetadataForReceivedMessage(null, hashCode, messageType)
+
+        then:
+        1 * mockLogger.logWarning(_)
+    }
+
+    def "savePartnerMetadataForReceivedMessage logs error and continues when updateMetadataForReceivedMessage throws error"() {
+        given:
+        def hashCode = new Random().nextInt()
+        def messageType = PartnerMetadataMessageType.RESULT
+        def receivedSubmissionId = "receivedId"
+        TestApplicationContext.injectRegisteredImplementations()
+        mockOrchestrator.updateMetadataForReceivedMessage(receivedSubmissionId, _ as String, messageType) >> { throw new PartnerMetadataException("Error") }
```

Comment:
Typically you want to put any mock behavior before you inject the registered implementations